Tema4/src/Animal_Product.java

```diff
@@ -0,0 +1,45 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Animal_Product extends Product implements Validity {
```

Comment:
Product can implement 2 interfaces, both Buyable and Validity so you can have that common functionality there